elastic/kibana
# [Vega] Restores signal values on refresh

Closes: #88976

**Describe the bug:** When the function kibanaAddFilter() is called in a Vega specification it will result in all signal values being reverted back to default value.

**Steps to reproduce:**
```
{
  $schema: https://vega.github.io/schema/vega/v5.json
  config: {
    kibana: {
      ## Important: this option should be true
      restoreSignalValuesOnRefresh: true
    }
  }
  signals: [
    {
      name: test
      value: default
      on: [
        {
          events: mousemove
          update: now()
        }
      ]
    }
    {
      name: test2
      value: default
      on: [
        {
          events: click
          update: kibanaAddFilter()
        }
      ]
    }
  ]
}
```

1. Move mouse over canvas to allow `test` signal to update to current time.
2. Click on canvas to trigger `kibanaAddFilter`.
3. Observe `test` signal reverted back to default value of `default`.

**Expected behavior:**

The function `kibanaAddFilter` should not reset all signal values back to default.

Screens: 

https://user-images.githubusercontent.com/20072247/107371673-98526300-6af5-11eb-9ff4-2ec3cac5c38a.mov
